applications/tari_dan_wallet_daemon/src/handlers/accounts.rs

```diff
@@ -69,9 +69,15 @@ const LOG_TARGET: &str = "tari::dan_wallet_daemon::handlers::transaction";
 
 pub async fn handle_create(
     context: &HandlerContext,
+    token: Option<String>,
     req: AccountsCreateRequest,
 ) -> Result<AccountsCreateResponse, anyhow::Error> {
     let sdk = context.wallet_sdk();
+    let jwt = sdk.jwt_api();
+    let permissions = jwt.get_permissions(token)?;
```

Comment:
```suggestion
    check_auth(&sdk, token, &[JrpcPermission::Admin])?;
    
    // Elsewhere
    fn check_auth(wallet_sdk: &WalletSdk, token: Option<String>, required_perms: &[JrpcPermission]) -> Result<JrpcPermissions, JsonRpcError> {
        let token = token.ok_or_else(|| unauthorized("no JWT token provided"))?;
        let permissions = jwt.get_permissions(token).map_err(|e| unauthorized(&format!("malformed  JWT: {e}")))?;
        for p in required_perms {
        if !permissions.has_permission(p) {
￼        return Err( unauthorized(&format!("Insufficient permissions. Required {p}"))));
￼    }
}
    }
    
    fn unauthorized(details: &str) -> JsonRpcError {
      JsonRpcError::new(
        JsonRpcErrorReason::ApplicationError(401),
            format!("Not authorized: {details}")
        serde_json::Value::Null,
    )
     }
    
```

Reply:
I did it slightly differently, now I get all JwtApiErrors as 401 (e.g. expiration of the token)